source/ImpostorPage.cpp

```diff
@@ -445,22 +445,22 @@ void ImpostorTexture::updateMaterials()
 ImpostorTexture::~ImpostorTexture()
 {
 	//Delete textures
-	assert(!texture.isNull());
+	assert(texture);
 	String texName(texture->getName());
 		
-	texture.setNull();
+	texture.reset();
 	if (TextureManager::getSingletonPtr())
-		TextureManager::getSingleton().remove(texName);
+		TextureManager::getSingleton().remove(texName, RGN_AUTODETECT);

```

Comment:
we could actually just remove it by ptr instead of by name. But your change certainly improves things